src/odemis/driver/technolution.py

```diff
@@ -284,127 +288,154 @@ def _assembleCalibrationData(self):
+        # Round to prevent uneven
+        total_line_scan_time = numpy.round(line_scan_time + flyback_time, 9)
+        calibration_frequency = 1 / total_line_scan_time
+
+        # Calculate time stamps and setpoints for one line scan (with sampling period is the descanner clock period)
+        # Round for floating point errors
+        nmr_setpoints = numpy.round(total_line_scan_time / descanner.clockPeriod.value, 10).astype(int)
+        time_points_descanner = numpy.linspace(0, total_line_scan_time, nmr_setpoints)
+        x_descan_setpoints = descan_offset_bits[0] + descan_amplitude_bits[0] * numpy.sin(
+                2 * math.pi * calibration_frequency * time_points_descanner)
+        # y_descan_setpoints should have a constant and zero output
+        y_descan_setpoints = 0 * time_points_descanner
+
+        # Calculate time stamps and setpoints for the scanner for one line scan
+        def calc_scanner_sampling_period(total_line_scan_time, scanner_clock_period, descanner_clock_period):
+            """
+            For creation of the calibration setpoints the sampling period is variable but the total number of
+            setpoints is limited to maximum 9000 points. Depending on the total line scanning time the calibration
+            setpoints have a variable sampling period (calibration dwell time) which is a multiple of the scanner
+            clock period. The number of setpoints (and the corresponding sampling period is chosen such that the
+            biggest number of setpoints is used (highest resolution of the signal).
+            The sampling period is defined by finding the higest number of setpoints for the calibration signal
+            with a sampling period equal to a whole multiple of the scanner clock period. This is done by iteratively
+            increasing the sampling period, via a multiplication, after which is checked if this would result in a
+            whole number of points to fit the total line scanning time.
+
+            :param total_line_scan_time:
+            :param scanner_clock_period: of the ASM
+            :return: (int) Sampling period in ticks
+                     (float) Sampling period is seconds
+                     (int) Number of sampling pointsI can already make a separate PR so you can check the
+            """
+            # TODO MAX_NMBR_POINT value of 9000 is not working for the entire range of the dwell time, this value
+            #  needs to be increased.
+            MAX_NMBR_POINT = 9000  # Constant maximum number of setpoints
+
+            # The minimal multiplication of the scanner clock period to get a sampling period with max 9000 setpoints
+            min_multiplication = int(numpy.ceil((total_line_scan_time / MAX_NMBR_POINT) / scanner_clock_period))
+            # Make sure the multiplication with the scanner clock period ensures the sampling period is bigger than
```

Comment:
-> `# If the found minimum sampling period (in ticks) is smaller than the minimal dwell time for the scanner, choose the minimum dwell time.`